zap/src/test/java/org/parosproxy/paros/network/HttpRequestHeaderUnitTest.java

```diff
@@ -204,42 +209,140 @@ void shouldReplaceAnyCookieHeaderWhenSettingCookieParams() {
+        return falseTestUrls("css", "css");
+    }
+
     @ParameterizedTest
-    @ValueSource(
-            strings = {
-                "http://example.org/css/file.ext", // In directory path
-                "http://ericsson.com/", // In domain name
-                "https://example.css", // In domain extension (TLD)
-                "https://example.css/dir/file.ext", // In domain extension (TLD)
-                "https://example.org/dir/file?foo=bar&thing=css", // In parameter value
-                "http://example.org/css/file.ext?foo=bar&type=.css", // In parameter value including
-                // period
-                "http://example.org/css/file.ext?foo=bar&thing=styles.css", // In parameter value,
-                // plausible filename
-                "https://example.org/dir/file?foo=bar&css=file.ext" // In parameter name
-            })
+    @MethodSource("falseTestCssUrls")
     void isCssShouldReturnFalseWhenUrlDoesNotIndicateCss(String url) {
         // Given
         HttpRequestHeader reqHeader = createRequestHeader(url);
         // When / Then
         assertFalse(reqHeader.isCss());
     }
 
+    private static Stream<Arguments> trueTestCssUrls() {
+        String[] exts = {"css"};
+        return trueTestUrls(exts);
+    }
+
     @ParameterizedTest
-    @ValueSource(
-            strings = {
-                "http://example.org/styles.css", // In path
-                "http://example.org/assets/css/styles.css", // In deeper path
-                "http://example.org/css/styles.css?foo=bar", // In path, ignoring params
-                "http://example.org/css/styles.css?foo=bar&thing=.css", // In path, ignoring params
-            })
+    @MethodSource("trueTestCssUrls")
     void isCssShouldReturnTrueWhenUrlIndicatesCss(String url) {
         // Given
         HttpRequestHeader reqHeader = createRequestHeader(url);
         // When / Then
         assertTrue(reqHeader.isCss());
     }
 
+    private static Stream<Arguments> falseTestImageUrls() {
+        return falseTestUrls("image", "gif");
+    }
+
+    @ParameterizedTest
+    @MethodSource("falseTestImageUrls")
+    void isImageShouldReturnFalseWhenUrlDoesNotIndicateImage(String url) {
+        // Given
+        HttpRequestHeader reqHeader = createRequestHeader(url);
+        // When / Then
+        assertFalse(reqHeader.isImage());
+    }
+
+    private static Stream<Arguments> trueTestImageUrls() {
+        // Per HttpRequestHeader.patternImage
+        String[] exts = {"bmp", "ico", "jpg", "jpeg", "gif", "tiff", "tif", "png", "svg"};
+        return trueTestUrls(exts);
+    }
+
+    @ParameterizedTest
+    @MethodSource("trueTestImageUrls")
+    void isImageShouldReturnTrueWhenUrlIndicatesImage(String url) {
+        // Given
+        HttpRequestHeader reqHeader = createRequestHeader(url);
+        // When / Then
+        assertTrue(reqHeader.isImage());
+    }
+
+    private static Stream<Arguments> falseTestUrls(String pathPart, String extension) {
```

Comment:
I wasn't sure on the name of `pathPart`, it's often the path but not always. I had bothered to include it as the response header check use those strings. Maybe it isn't worth it? Obviously separate code 🤷

Reply:
I don't think it's necessary since it's not relevant for the code being tested.

Reply:
Okay I’ll simplify it and remove those parts